src/odemis/gui/cont/tabs/sparc2_chamber_tab.py

```
@@ -220,33 +225,50 @@ def _on_switch_btn(self, evt):
-            logging.warning("Axes order of mirror is %s, while should have s and l axes",
-                            axes_order)
+        pos_parked = get_mirror_pos_parked(mirror)
+        mirror_axes = set(mirror.axes.keys())
+        mirror_md = mirror.getMetadata()
+        if mirror_axes == MIRROR_AXES_LS:
+            axes_order = mirror_md.get(model.MD_AXES_ORDER_REF, ("s", "l"))
+        else:
+            try:
+                axes_order = mirror_md[model.MD_AXES_ORDER_REF]
+            except KeyError:
+                raise ValueError("Mirror actuator has no metadata AXES_ORDER_REF")
```

Ruff (0.15.21)

[warning] Within an `except` clause, raise exceptions with `raise ... from err` or `raise ... from None` to distinguish them from errors in exception handling

(B904)